app.py

```diff
@@ -30,45 +32,141 @@ def init_app(self):
+
+        :param node: RenderMan node
+        :param network: Network type
+        """
         self.handler.copy_to_clipboard(node, network)
 
     @staticmethod
     def get_all_renderman_nodes():
-        # Get all nodes from node type sgtk_hdprman
-        lop_nodes = (
-            hou.lopNodeTypeCategory().nodeType("sgtk_hdprman").instances()
-        )
+        """Get all nodes from node type sgtk_hdprman"""
+        lop_nodes = hou.lopNodeTypeCategory().nodeType("sgtk_hdprman").instances()
         rop_nodes = hou.ropNodeTypeCategory().nodeType("sgtk_ris").instances()
         nodes = lop_nodes + rop_nodes
         return nodes
 
-    @staticmethod
-    def get_output_path(node):
+    def get_output_path(self, node, aov_name, network="rop"):
+        """Calculate render path for an aov
+
+        :param node: RenderMan node
+        :param aov_name: AOV name
+        :param network: Network type
+        """
+        current_filepath = hou.hipFile.path()
+
+        work_template = self.get_template("work_file_template")
+        render_template = self.get_template("output_render_template")
+
+        resolution_x_field = "resolutionx"
+        resolution_y_field = "resolutiony"
+
+        resolution_x = 0
+        resolution_y = 0
+
+        evaluate_parm = True
+
+        # Because RenderMan in the rop network uses different
+        # parameter names, we need to change some bits
+        if network == "rop":
+            camera = node.parm("camera").eval()
+
+            evaluate_parm = False
+            resolution_x = hou.node(camera).parm("resx").eval()
+            resolution_y = hou.node(camera).parm("resy").eval()
+
+            if node.parm("override_camerares").eval():
+                res_fraction = node.parm("res_fraction").eval()
+
+                if res_fraction == "specific":
+                    evaluate_parm = True
+                    resolution_x_field = "res_overridex"
+                    resolution_y_field = "res_overridey"
+
+                else:
+                    resolution_x = resolution_x * res_fraction
+                    resolution_y = resolution_y * res_fraction
+
+        # Set fields
+        fields = work_template.get_fields(current_filepath)
+        fields["SEQ"] = "FORMAT: $F"
+        fields["output"] = node.parm("name").eval()
+        fields["aov_name"] = aov_name
+        if evaluate_parm is True:
+            fields["width"] = node.parm(resolution_x_field).eval()
+            fields["height"] = node.parm(resolution_y_field).eval()
 
-        if node.type().nameComponents()[2] == "sgtk_ris":
-            parameter = "ri_display_0"
         else:
-            parameter = "picture"
+            fields["width"] = resolution_x
+            fields["height"] = resolution_y
+
+        return render_template.apply_fields(fields).replace(os.sep, "/")
 
-        output_path = node.parm(parameter).eval()
-        return output_path
+    def get_metadata_config(self):
```

Comment:
Is this method used anywhere outside of this object?

Else it would be better to make this private. This will show any developers that this method is only required inside this object, but not outside.

There are two ways of making things private. Either adding a single underscore or a double underscore. There is a difference between the two. 

The first one, for example _get_metadata_config is a convention. This shows the other developer that its not meant to be used outside. IDE's also recognize this.

The second option is super private. This would be __get_metadata_config. This is not only a convention, its not possible to use this method outside as Python renames it to an anonymous name.

The second option is rarely useful, only use this option if your code would break if this method is used. So for here, I would suggest to rename this to _get_metadata_config

Reply:
Both mentions of the methods that should be private are called from the PythonModule on the HDAs, and therefore Mervind and I do not consider it private.